lib/active_record/typed_store/dsl.rb

```diff
@@ -1,29 +1,47 @@
+require 'active_record/typed_store/column'
 module ActiveRecord::TypedStore
-
   class DSL
+    attr_reader :columns, :coder
 
-    attr_reader :columns
-
-    def initialize(accessors=true)
-      @accessors = accessors
-      @columns = []
+    def initialize(options)
+      @coder = options.fetch(:coder, default_coder)
+      @columns = {}
       yield self
     end
 
+    def default_coder
+      ActiveRecord::Coders::YAMLColumn.new
+    end
+
     def accessors
-      @columns.select(&:accessor?).map(&:name)
+      @columns.values.select { |v| v.accessor }.map(&:name)
     end
 
-    [:string, :text, :integer, :float, :datetime, :date, :boolean, :any].each do |type|
-      define_method(type) do |name, options = {}|
-        @columns << Column.new(name, type, {accessor: @accessors}.merge(options))
+    delegate :keys, to: :@columns
+
+    NO_DEFAULT_GIVEN = Object.new
+    [:string, :text, :integer, :float, :datetime, :date, :boolean, :decimal, :any].each do |type|
+      define_method(type) do |name, **options|
+        if options.key?(:default)
+          options[:default] = decode_default(options[:default])
+        end
+        @columns[name] = Column.new(name, type, options)
       end
     end
-
-    def decimal(name, options = {})
-      @columns << Column.new(name, :decimal, {accessor: @accessors, limit: 20, scale: 6}.merge(options))
+    alias_method :date_time, :datetime
+
+    private
+
+    def decode_default(value)
+      if @coder.is_a?(ActiveRecord::Coders::YAMLColumn)
+        begin
+          @coder.load(value)
+        rescue
```

Comment:
Oh, didn't noticed that rescue on the first review. What kind of error do you expect to catch here? Feels bad to rescue blindly like that.